scout/server/blueprints/variants/controllers.py

```diff
@@ -88,17 +88,29 @@ def variants(store, institute_obj, case_obj, variants_query, page=1, per_page=50
             evaluations.append(evaluation_obj)
         variant_obj["evaluations"] = evaluations
 
+        if variant_obj.get("dismiss_variant") is not None:
```

Comment:
It works! Was there any particular reason why you created this code in the variants controller? Because an alternative could have been to pass the DISMISS_VARIANT_OPTIONS along and to handle the situation in the template, as it's done for MANUAL_RANK_OPTIONS, CANCER_TIER_OPTIONS, SEVERE_SO_TERMS

Reply:
😄 Well, no, not more than that I followed my last version of this from the research side (compare https://github.com/Clinical-Genomics/scout/blob/ea2136e34094351d98b05c8e180850abb2c50839/scout/server/blueprints/variants/controllers.py#L146), where it is generated in a controller subroutine.

I'm all for plurality, but if I have to choose between the four-five different ways of doing essentially the same things we use, I do like the version where data for display is prepared in the controller over passing constants all the way to jinja. Don't get me wrong: jinja is cool, they all work, and repetitions tend to happen in all.

Reply:
I agree with you that stuff like this is better handled in the controllers but I also think that it would be best to have a consistent way to populate all these badges (with MANUAL_RANK_OPTIONS, CANCER_TIER_OPTIONS, SEVERE_SO_TERMS, DISMISS_VARIANT_OPTIONS). Would it be a lot of work to have a unique function (outside the variants views) that handles these variables and send stuff to every variantS template?

Reply:
No, I don't think so, but they are used in several places, so I worried about missing something and introducing errors. But I do agree - and in its own way I also think CodeFactor agrees. Breaking out a few things of the controllers and templates would not hurt. I can give it a start and see if I run into any problems.

Reply:
At least locally in cell_rank, it's just manual_rank that uses the lookup table directly in the template, the rest use controller (sic) based prepared strings. In the cancer case again  manual_rank as well as the analogous cancer_rank. I have a version now, but will sleep on it: in particular it touches the subtlety with how research classifications are different than clinical ones yet use the same display - but asymmetric since clinical should not be informed by research.

Reply:
ok, let me know when you are ready and I can review!